src/openrct2/actions/GuestSetNameAction.h

```diff
@@ -9,7 +9,7 @@
 
 #pragma once
 
-#include "../world/Sprite.h"
+#include "../rct12/RCT12.h"
```

Comment:
Why is this dependent on RCT12.h? That should only be the case for import/export code.

Reply:
I moved `SPRITE_INDEX_NULL` and `MAX_SPRITES` in there

Reply:
Well neither of those should be in the rct12 headers. They are openrct2 variables. I would undo that change.

Reply:
This makes up for a significant chunk of this PR, so I'd like to keep them somewhere else than `Sprite.h`. Do you suggest any other file or should I create another one?

Reply:
The sprite null is such a common constant i would maybe move it all the way up into common.h. Max sprites should only be used in import/export but i know at present we use it when validating as well and for a good few loops. Perhaps we should just make a isEntityIdValid function instead for the validation uses

Reply:
`MAX_SPRITES` is used in quite a few more places: https://github.com/OpenRCT2/OpenRCT2/search?q=MAX_SPRITES

```
src/openrct2-ui/windows/EditorBottomToolbar.cpp:289:            || (GetNumFreeEntities() == MAX_SPRITES && !(gParkFlags & PARK_FLAGS_SPRITES_INITIALISED)))
src/openrct2-ui/windows/EditorBottomToolbar.cpp:349:            if (GetNumFreeEntities() != MAX_SPRITES || gParkFlags & PARK_FLAGS_SPRITES_INITIALISED)
src/openrct2-ui/windows/EditorBottomToolbar.cpp:374:    else if (GetNumFreeEntities() != MAX_SPRITES)
src/openrct2/Editor.cpp:312:        for (int32_t i = 0; i < MAX_SPRITES; i++)
src/openrct2/Game.cpp:591:    for (size_t i = 0; i < MAX_SPRITES; i++)
src/openrct2/GameStateSnapshots.cpp:148:            [](const size_t index) { return reinterpret_cast<rct_sprite*>(GetEntity(index)); }, MAX_SPRITES, true);
src/openrct2/GameStateSnapshots.cpp:174:        spriteList.resize(MAX_SPRITES);
src/openrct2/GameStateSnapshots.cpp:182:        snapshot.SerialiseSprites([&spriteList](const size_t index) { return &spriteList[index]; }, MAX_SPRITES, false);
src/openrct2/actions/GuestSetNameAction.cpp:59:    if (_spriteIndex >= MAX_SPRITES)
src/openrct2/actions/PeepPickupAction.cpp:38:    if (_spriteId >= MAX_SPRITES || _spriteId == SPRITE_INDEX_NULL)
src/openrct2/actions/StaffFireAction.cpp:33:    if (_spriteId >= MAX_SPRITES)
src/openrct2/actions/StaffSetCostumeAction.cpp:58:    if (_spriteIndex >= MAX_SPRITES)
src/openrct2/actions/StaffSetNameAction.cpp:43:    if (_spriteIndex >= MAX_SPRITES)
src/openrct2/actions/StaffSetOrdersAction.cpp:39:    if (_spriteIndex >= MAX_SPRITES)
src/openrct2/actions/StaffSetPatrolAreaAction.cpp:35:    if (_spriteId >= MAX_SPRITES)
src/openrct2/interface/InteractiveConsole.cpp:1268:    console.WriteFormatLine("Sprites: %d/%d", spriteCount, MAX_SPRITES);
src/openrct2/interface/InteractiveConsole.cpp:1591:    for (int i = 0; i < MAX_SPRITES; i++)
src/openrct2/interface/Window.cpp:2144:    if (spriteIndex < MAX_SPRITES || spriteIndex == SPRITE_INDEX_NULL)
src/openrct2/rct1/S4Importer.cpp:1358:        for (size_t i = 0; i < MAX_SPRITES; i++)
src/openrct2/scripting/ScMap.hpp:53:            return MAX_SPRITES;
src/openrct2/scripting/ScMap.hpp:86:            if (id >= 0 && id < MAX_SPRITES)
src/openrct2/world/Sprite.cpp:29:static rct_sprite _spriteList[MAX_SPRITES];
src/openrct2/world/Sprite.cpp:33:static bool _spriteFlashingList[MAX_SPRITES];
src/openrct2/world/Sprite.cpp:131:    return spriteIndex >= MAX_SPRITES ? nullptr : &_spriteList[spriteIndex].misc;
src/openrct2/world/Sprite.cpp:140:    openrct2_assert(spriteIndex < MAX_SPRITES, "Tried getting sprite %u", spriteIndex);
src/openrct2/world/Sprite.cpp:261:    for (int32_t i = 0; i < MAX_SPRITES; ++i)
src/openrct2/world/Sprite.cpp:293:    for (size_t i = 0; i < MAX_SPRITES; i++)
src/openrct2/world/Sprite.cpp:323:        for (size_t i = 0; i < MAX_SPRITES; i++)
src/openrct2/world/Sprite.cpp:948:    assert(sprite->sprite_index < MAX_SPRITES);
src/openrct2/world/Sprite.cpp:954:    assert(sprite->sprite_index < MAX_SPRITES);
src/openrct2/world/Sprite.h:22:#define MAX_SPRITES 10000
test/testpaint/Compat.cpp:173:    assert(sprite_idx < MAX_SPRITES);
test/tests/S6ImportExportTests.cpp:40:    rct_sprite sprites[MAX_SPRITES];
test/tests/S6ImportExportTests.cpp:109:    for (size_t spriteIdx = 0; spriteIdx < MAX_SPRITES; spriteIdx++)
test/tests/S6ImportExportTests.cpp:464:    for (size_t spriteIdx = 0; spriteIdx < MAX_SPRITES; ++spriteIdx)
```

Some in areas that wouldn't be compatible with `isEntityIdValid`, but that sounds like it could be moved to the new `Entity.h`. Should it also get renamed to `MAX_ENTITIES`?

Reply:
Yes should be renamed and should be in entity.h or really entitymanager.h but we don't have entity manager yet. The editor users of this should really be modified. I'll add that to my never ending list.

Reply:
All done now.